src/nvim/eval/typval_encode.c.h

```diff
@@ -391,11 +391,7 @@ static int _TYPVAL_ENCODE_CONVERT_ONE_VALUE(
     }
     break;
   case VAR_SPECIAL:
-    switch (tv->vval.v_special) {
-    case kSpecialVarNull:
-      TYPVAL_ENCODE_CONV_NIL(tv);  // -V1037
-      break;
-    }
+    TYPVAL_ENCODE_CONV_NIL(tv);  // -V1037
```

Comment:
I feel this can cause confusion if more values are added to `SpecialVarValue` later.

Reply:
I agree. The only way this can be done is a refactor that consistently removes `tv->vval.v_special` everywhere. But the complexity savings are negligee, so better to keep this to be open for more special values, which might happen.